src/components/SearchPhotos/SearchPhotos.js

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import axios from "axios";
+import LoadingBar from "react-top-loading-bar";
+import { Container, StyledLink, StyledImage } from "./searchPhotos.styles";
+
+class SearchPhotos extends React.Component {
+  state = {
+    photoData: null,
+    isLoading: false,
+  };
+
+  loadingBar = React.createRef();
+
+  baseUrl = `${process.env.REACT_APP_API_BASE_URL}/search`;
+
+  getPhotoData = async (searchTerm) => {
+    try {
+      this.loadingBar.current.continuousStart();
+      this.setState({ isLoading: true });
+      const { data } = await axios(
+        `${this.baseUrl}/photos?query=${searchTerm}&client_id=${process.env.REACT_APP_API_KEY}`
+      );
+      this.setState({ photoData: data, isLoading: false });
+      this.loadingBar.current.complete();
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  componentDidMount() {
+    const { searchTerm } = this.props;
+    this.getPhotoData(searchTerm);
+    this.setState({ searchTerm });
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { searchTerm } = this.props;
+    if (prevProps.searchTerm !== searchTerm) {
+      this.getPhotoData(searchTerm);
+      this.setState({ searchTerm });
```

Comment:
Second time i see this i guess it creates a new property in state and it sets it equal to its same value right? is it beacuse they have the same name right?

Reply:
good catch, do you need to save this is state if you already have it in props?

Reply:
Yes, because I am pulling information from state in the render so I would need to update state with what I get from props, right?